File: proiectSe/src/app/model/recipe-dto.ts
export class RecipeDTO {
  name: string | null = null;
  description: string | null = null;
  ingredients: string[] = [];
  duration: string | null = null;
  steps: string[] = [];
  calories: string | null = null;
  fat: string | null = null;
  sugar: string | null = null;
  protein: string | null = null;
  sodium: string | null = null;
  carbohydrates: string | null = null;
}

File: demo/src/main/java/com/example/demo/Domain/RecipeDTO.java
package com.example.demo.Domain;

public class RecipeDTO {
    String name;
    String description;
    String ingredients;
    String steps;
    String duration;

    public RecipeDTO(){

    }


    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getIngredients() {
        return ingredients;
    }

    public void setIngredients(String ingredients) {
        this.ingredients = ingredients;
    }

    public String getSteps() {
        return steps;
    }

    public void setSteps(String steps) {
        this.steps = steps;
    }

    public String getDuration() {
        return duration;
    }

    public void setDuration(String duration) {
        this.duration = duration;
    }
}
